scripts/validateTests.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+import { IAssignmentTestCase } from './types';
+
+
+const parseJSON = (testFilePath: string) => {
+  try {
+    const fileContents = fs.readFileSync(testFilePath, 'utf-8');
+    const parsedJSON = JSON.parse(fileContents);
+    return parsedJSON as IAssignmentTestCase;
+  } catch (err) {
+    console.error(`failed to parse JSON in ${testFilePath}`)
+    console.error(err);
+    process.exit(1);
+  }
+}
+
+const validateAssignmentDetails = (testFilePath: string, testCase: IAssignmentTestCase) => {
+  const errors: string[] = [];
+  testCase.subjects.forEach(({ subjectKey, assignment, assignmentDetails }) => {
+    const addError = (message: string) => {
+      errors.push(`Error (subjectKey "${subjectKey}"): ${message}`);
+    }
+    // stringify for deep equality checks with objects
+    const stringifiedDefaultValue = JSON.stringify(testCase.defaultValue);
+    const stringifiedAssignment = JSON.stringify(assignment);
+    const stringifiedDetailsValue = JSON.stringify(assignmentDetails.value);
+    const stringifiedDetailsVariationValue = JSON.stringify(assignmentDetails.variationValue);
```

Comment:
another benefit of using jest is you could use a deep object equality instead of having to stringify

Reply:
updated!